### PR TITLE
Be more permissive while parsing parameter lists

### PR DESCRIPTION
**Pull request by na-g**
Allows spaces between parameters
Allows multiple inner semicolons
Allows leading and following semicolons

Adds related tests.
95% of this patch is the indention of a large block.

### CODE REVIEW

File: src/attribute_type.rs [hunk 1]
```diff
@@ -1905,146 +1905,153 @@ fn parse_fmtp(to_parse: &str) -> Result<SdpAttribute, SdpParserInternalError> {
...
-    if parameter_token.contains('=') {
-        let parameter_tokens: Vec<&str> = parameter_token.split(';').collect();
-
-        for parameter_token in parameter_tokens.iter() {
-            let name_value_pair: Vec<&str> = parameter_token.splitn(2, '=').collect();
-
-            if name_value_pair.len() != 2 {
-                return Err(SdpParserInternalError::Generic(
-                    "A fmtp parameter must be either a telephone event, a parameter list or
+    for parameter_token in tokens[1..].to_vec() {
```
Comment (nils-ohlmeier) [inline src/attribute_type.rs]: From a vec to a slice back to a vec again doesn't seem very efficient.

I think converting this from the initial collect() call to using the next() call on the iterator would be better.

File: src/attribute_type.rs [hunk 2]
```diff
@@ -1905,146 +1905,153 @@ fn parse_fmtp(to_parse: &str) -> Result<SdpAttribute, SdpParserInternalError> {
...
+            // Permit Leading/Trailing/Inner ';' by filtering out empty splits
+            let parameter_tokens: Vec<&str> = parameter_token
+                .split(';')
+                .filter(|token| !token.is_empty())
+                .collect();
+            for parameter_token in parameter_tokens.iter() {
+                let name_value_pair: Vec<&str> = parameter_token.splitn(2, '=').collect();
+                if name_value_pair.len() != 2 {
+                    return Err(SdpParserInternalError::Generic(
+                        "A fmtp parameter must be either a telephone event, a parameter list or
```
Comment (nils-ohlmeier) [inline src/attribute_type.rs]: The content of this error doesn't look right.
If I remember right the formatting of the indented string "a red codec list" also appeared indented in the file_parser output.

Comment (na-g) [reply src/attribute_type.rs]: Yeah, I remember seeing that as well...
Fixed.

File: src/attribute_type.rs (hunk 1 above)
Comment (na-g) [reply src/attribute_type.rs]: Fixed. I changed the to_vec() to iter() as to avoid the use of next().

**Pull request merged**